src/EditorFeatures/Core/InlineParameterNameHints/InlineParameterNameHintsDataTaggerProvider.cs

```diff
@@ -53,7 +53,25 @@ internal class InlineParameterNameHintsDataTaggerProvider : AsynchronousTaggerPr
         protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
         {
             // TaggerDelay is NearImmediate because we want the renaming and tag creation to be instantaneous
-            return TaggerEventSources.OnWorkspaceChanged(subjectBuffer, TaggerDelay.NearImmediate, _listener);
+            return TaggerEventSources.Compose(
+                TaggerEventSources.OnViewSpanChanged(ThreadingContext, textViewOpt, textChangeDelay: TaggerDelay.Short, scrollChangeDelay: TaggerDelay.NearImmediate),
+                TaggerEventSources.OnWorkspaceChanged(subjectBuffer, TaggerDelay.NearImmediate, _listener));
+        }
+
+        protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView textView, ITextBuffer subjectBuffer)
+        {
+            this.AssertIsForeground();
+
+            // Find the visible span some 100 lines +/- what's actually in view.  This way
+            // if the user scrolls up/down, we'll already have the results.
+            var visibleSpanOpt = textView.GetVisibleLinesSpan(subjectBuffer, extraLines: 100);
+            if (visibleSpanOpt == null)
+            {
+                // Couldn't find anything visible, just fall back to classifying everything.
```

Comment:
Comment needs updating since this isn't classification here.

Reply:
Sorry forgot to update that comment, fixed.